numba/tests/test_comprehension.py

```diff
@@ -0,0 +1,256 @@
+from __future__ import print_function
```

Comment:
I think it'll actually make sense to add more variants to this file when other comprehensions are becoming supported, since they are all comprehensions, and they more or less will exercise the same inline closure pass. 

I'm actually adding tests for "array comprehension" (turn `array([...])` into direct array creation without intermediate list) to this file in an upcoming PR :-)